### PR TITLE
Implement SassColor.

### PR DESCRIPTION
**Pull request by Awjin**
https://github.com/sass/sass-spec/pull/1729

### CODE REVIEW

File: lib/src/value/color.ts [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    }
+    return this.lightnessInternal!;
+  }
+
+  /** `this`'s whiteness value. */
+  get whiteness(): number {
+    // Because HWB is (currently) used much less frequently than HSL or RGB, we
+    // don't cache its values because we expect the memory overhead of doing so
+    // to outweigh the cost of recalculating it on access.
+    return (Math.min(Math.min(this.red, this.green), this.blue) / 255) * 100;
```
Comment (nex3) [inline lib/src/value/color.ts:191]:
```suggestion
    return Math.min(this.red, this.green, this.blue) / 255 * 100;
```

`Math.min()` takes any number of arguments. Same with `Math.max()` below.

File: lib/src/value/color.ts [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+  // `red`, `green`, and `blue`.
+  //
+  // Algorithm from https://en.wikipedia.org/wiki/HSL_and_HSV#RGB_to_HSL_and_HSV
+  private rgbToHsl(): void {
+    const scaledRed = this.red / 255;
+    const scaledGreen = this.green / 255;
+    const scaledBlue = this.blue / 255;
+
+    const max = Math.max(Math.max(scaledRed, scaledGreen), scaledBlue);
+    const min = Math.min(Math.min(scaledRed, scaledGreen), scaledBlue);
```
Comment (nex3) [inline lib/src/value/color.ts:331]: Same here

**Review by nex3 — APPROVED** [review]

**Comment by Awjin** [comment]
@nex3 I'm confused as to how type checks passed for Dart Sass, but are broken here. Shouldn't `ts-node` be catching these?

File: lib/src/value/color.ts [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    } else {
+      this.saturationInternal = (100 * delta) / (2 - max - min);
+    }
+  }
+
+  // Computes `this`'s red`, `green`, and `blue` channels based on `hue`,
+  // `saturation`, and `value`.
+  //
+  // Algorithm from the CSS3 spec: https://www.w3.org/TR/css3-color/#hsl-color.
+  private hslToRgb(): void {
```
[the remark this answers is not in the document]
Comment (nex3) [reply lib/src/value/color.ts:328]: Round-trips between the host and the compiler are very expensive, so we want to minimize them if at all possible. The color conversion algorithms are ultimately pretty simple, so I think it's worth asking wrappers to implement.

**Pull request merged**